src/main/java/org/folio/marc4ld/service/marc2ld/authority/MarcAuthority2ldMapperImpl.java

```diff
@@ -63,6 +71,26 @@ private Resource fillResource(Resource resource, org.marc4j.marc.Record marcReco
     marcRecord.getControlFields()
       .forEach(controlField -> authorityIdentifierProcessor.setIdentifier(resource, controlField));
     resource.setId(hashService.hash(resource));
+    folioMetadataFrom(marcRecord).ifPresent(resource::setFolioMetadata);
     return resource;
   }
+
+  private Optional<FolioMetadata> folioMetadataFrom(org.marc4j.marc.Record marcRecord) {
+    return marcRecord.getDataFields().stream()
+      .filter(dataField -> dataField.getTag().equals(FIELD_UUID))
+      .findFirst().map(
+        metadata -> new FolioMetadata()
+          .setSource(MARC)
+          .setInventoryId(getSubfieldValue(metadata, SUBFIELD_INVENTORY_ID))
+          .setSrsId(getSubfieldValue(metadata, S))
+      );
+  }
+
+  private String getSubfieldValue(DataField dataField, char code) {
```

Comment:
there is ```getSubfieldValue``` method in ```MarcUtil``` class
how about moving this code there?

Reply:
done.